api/src/integrations/epic/epic-datasource.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
+
+const QS = require('qs');
+const integrationUtils = require('../integration-utils');
+const EPIC_TYPE = require('./epic-type-enum');
+const defaultLog = require('../../utils/logger')('epic-integration');
+
+const MAX_PAGE_SIZE = Number.MAX_SAFE_INTEGER;
+
+class EpicDatasource {
+  /**
+   * Creates an instance of EpicDatasource.
+   *
+   * @param {TaskAuditRecord} taskAuditRecord an instance of TaskAuditRecord.
+   * @memberof EpicDatasource
+   */
+  constructor(taskAuditRecord) {
+    this.taskAuditRecord = taskAuditRecord;
+  }
+
+  /**
+   * Main function that runs all necessary operations to sync Epic records for a given type.
+   *
+   * @async
+   * @param {EPIC_TYPE} type type of Epic record to sync (required)
+   * @param {object} params params to filter which records to sync (optional)
+   * @returns {object} overall status of update process
+   * @memberof EpicDatasource
+   */
+  async syncRecords(type, params = {}) {
+    defaultLog.info(`syncRecords - type: ${type} - params: ${JSON.stringify(params)}`);
+
+    const status = {};
+
+    // Get record type specific utils (Order, Inspection, etc)
+    const typeUtils = this.getRecordTypeUtils(type);
+
+    // Build request url
+    const typeId = await this.getRecordTypeId(type); // TODO Need to account for milestone
+    const queryParams = { ...params, ...this.getBaseParams(typeId, MAX_PAGE_SIZE, 0) };
+    const url = this.getIntegrationUrl(this.getHostname(), this.getPathname(), queryParams);
+
+    await this.taskAuditRecord.updateTaskRecord({ dataSource: url });
+
+    // Get Epic records
+    const data = await integrationUtils.getRecords(url);
+    const epicRecords = data && data[0] && data[0].searchResults;
+
+    // Failed to find any epic records
+    if (!epicRecords || !epicRecords.length) {
+      defaultLog.info(`syncRecords - status: ${JSON.stringify(status)}`);
+
+      status.epicStatus = 'no epic records found';
+      return status;
+    }
+
+    // Update task record with number of epic records found
+    this.taskAuditRecord.updateTaskRecord({ itemTotal: epicRecords.length });
+
+    // Add epic meta to status (optional)
+    const epicMeta = data && data[0] && data[0].meta && data[0].meta[0];
```

Comment:
`const epicMeta = data && data[0] && data[0].meta && data[0].meta[0];`
to
`const epicMeta = data && data[0] && data[0].meta && data[0].meta.length;`